src/main/java/it/gov/pagopa/pu/fileshare/service/send/SendFileFacadeServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package it.gov.pagopa.pu.fileshare.service.send;
+
+import it.gov.pagopa.pu.fileshare.connector.send_notification.NotificationService;
+import it.gov.pagopa.pu.fileshare.exception.custom.FileUploadException;
+import it.gov.pagopa.pu.fileshare.exception.custom.InvalidFileException;
+import it.gov.pagopa.pu.fileshare.service.FileService;
+import it.gov.pagopa.pu.fileshare.service.FileStorerService;
+import it.gov.pagopa.pu.fileshare.service.UserAuthorizationService;
+import it.gov.pagopa.pu.fileshare.util.AESUtils;
+import it.gov.pagopa.pu.fileshare.util.FileUtils;
+import it.gov.pagopa.pu.p4paauth.dto.generated.UserInfo;
+import it.gov.pagopa.pu.sendnotification.dto.generated.LoadFileRequest;
+import it.gov.pagopa.pu.sendnotification.dto.generated.StartNotificationResponse;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.NoSuchAlgorithmException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Slf4j
+@Service
+public class SendFileFacadeServiceImpl implements SendFileFacadeService {
+
+  private final UserAuthorizationService userAuthorizationService;
+  private final FileService fileService;
+  private final FileStorerService fileStorerService;
+  private final NotificationService notificationService;
+  private final String sendFolder;
+  private final String archivedSubFolder;
+
+
+  public SendFileFacadeServiceImpl(
+    UserAuthorizationService userAuthorizationService, FileService fileService,
+    FileStorerService fileStorerService,
+    NotificationService notificationService,
+    @Value("${folders.send-file-folder}") String sendFolder,
+    @Value("${folders.process-target-sub-folders.archive}") String archivedSubFolder) {
+    this.userAuthorizationService = userAuthorizationService;
+    this.fileService = fileService;
+    this.fileStorerService = fileStorerService;
+    this.notificationService = notificationService;
+    this.sendFolder = sendFolder;
+    this.archivedSubFolder = archivedSubFolder;
+  }
+
+  @Override
+  public StartNotificationResponse uploadSendFile(Long organizationId, String sendNotificationId, String digest,
+    MultipartFile sendFile, UserInfo user, String accessToken) {
+    userAuthorizationService.checkUserAuthorization(organizationId, user, accessToken);
+    fileService.validateFile(sendFile);
+    String fileName = sendNotificationId+"_"+sendFile.getOriginalFilename();
+
+    try {
+      if(!digest.equals(FileUtils.calculateFileHash(sendFile.getInputStream())))
+        throw new InvalidFileException("Invalid digest");
+    } catch (IOException | NoSuchAlgorithmException e) {
+      throw new FileUploadException(e.getMessage());
+    }
```

Comment:
move into a private method `validateDigest`

Reply:
ps: have you tested this API?
I have a doubt that reading the stream here could lead to an already read stream: thus the save operation will create an empty file!

If this is your case, one solution could be calculate the digest while storing it with a wrapped InputStream